puppet-mode.el

```diff
@@ -1093,7 +1128,11 @@ for each entry."
   (setq-local beginning-of-defun-function #'puppet-beginning-of-defun-function)
   ;; Indentation
   (setq-local indent-line-function #'puppet-indent-line)
-  (setq indent-tabs-mode puppet-indent-tabs-mode)
+  (if puppet-use-smie
+      (smie-setup puppet-smie-grammar #'puppet-smie-rules
+                  :forward-token #'puppet--smie-forward-token
+                  :backward-token #'puppet--smie-backward-token)
+      (setq indent-tabs-mode puppet-indent-tabs-mode))
```

Comment:
Why did you move this into the `if`?  `indent-tabs-mode` should be independent of SMIE shouldn't it?